pkg/apis/admissionregistration/validation/validation.go

```diff
@@ -211,6 +211,7 @@ func validateAdmissionReviewVersions(versions []string, requireRecognizedAdmissi
 // ValidateValidatingWebhookConfiguration validates a webhook before creation.
 func ValidateValidatingWebhookConfiguration(e *admissionregistration.ValidatingWebhookConfiguration) field.ErrorList {
 	return validateValidatingWebhookConfiguration(e, validationOptions{
+		compileCEL:                              true,
```

Comment:
I feel like this should just be generalized to `ignoreMatchConditions`, and basically just skip validation of match conditions if they're unchanged. I'll defer to @liggitt on this though.

Reply:
so this checks for just "if there's no new expressions, then skip compiling them" since the compilation is expensive, it's possible match conditions get renamed or something and then we would recompile here. I can implement it however is preferred though.

Reply:
Skipping validation if nothing changed isn't just an optimization though. For example, suppose we wanted to increase the total number of match conditions - to handle version skew between API servers, it's better to ignore a potentially "invalid" object if nothing has changed.

Reply:
okay I'll implement it this way!

Reply:
pushed the fix, let me know if that's better